### PR TITLE
[Feature](generic-aggregation) add some type define for generic aggregate functions support

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes

add some type define for generic aggregate functions support
part of https://github.com/apache/doris/pull/18541
## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

File: be/src/vec/data_types/data_type_agg_state.h [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+} // namespace vectorized
+} // namespace doris
+
+namespace doris::vectorized {
+
+class DataTypeAggState : public DataTypeString {
+public:
+    const char* get_family_name() const override { return "AggState"; }
+
+    TypeIndex get_type_id() const override { return TypeIndex::String; }
```
Comment (HappenLee) [inline be/src/vec/data_types/data_type_agg_state.h:46]: why not we add a new TypeIndex？

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run arm

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**